securing_apps/topics/saml/java/multi-tenancy.adoc

```diff
@@ -0,0 +1,72 @@
+[[_saml_multi_tenancy]]
+==== Multi Tenancy
+
+SAML offers the same functionality than OIDC to <<_multi_tenancy,Multi Tenancy>>. It means that a single target application (WAR) can be secured with multiple {project_name} realms. The realms can be located on the same {project_name} instance or on different instances.
+
+In practice, this means that the application needs to have multiple `keycloak-saml.xml` adapter configuration files.
+
+You could have multiple instances of your WAR with different adapter configuration files deployed to different context-paths. However, this may be inconvenient and you may also want to select the realm based on something else than context-path.
+
+{project_name} makes it possible to have a custom config resolver so you can choose what adapter config is used for each request. In SAML the configuration is only interesting in the login processing, once the user is logged in, the session is authenticated and it does not matter if the `keycloak-saml.xml` returned is different. For that reason, returning the same configuration for the same session is the correct way to go.
+
+To achieve this first you need to create an implementation of `org.keycloak.adapters.saml.SamlConfigResolver`. For example, the following example uses the `Host` header to locate the proper configuration and loads it and the associated elements from the applications's java classpath:
+
```

Comment:
SAML offers the same functionality as OIDC for <<_multi_tenancy,Multi Tenancy>>, meaning that a single target application (WAR) can be secured with multiple {project_name} realms. The realms can be located on the same {project_name} instance or on different instances.

To do this, the application must have multiple `keycloak-saml.xml` adapter configuration files.

While you could have multiple instances of your WAR with different adapter configuration files deployed to different context-paths., this may be inconvenient and you may also want to select the realm based on something other than context-path.

{project_name} makes it possible to have a custom config resolver, so you can choose which adapter config is used for each request. In SAML, the configuration is only interesting in the login processing; once the user is logged in, the session is authenticated and it does not matter if the `keycloak-saml.xml` returned is different. For that reason, returning the same configuration for the same session is the correct way to go.

To achieve this, create an implementation of `org.keycloak.adapters.saml.SamlConfigResolver`. The following example uses the `Host` header to locate the proper configuration and load it and the associated elements from the applications's Java classpath: